src/Engine/ProtoCore/Core.cs

```diff
@@ -315,53 +324,59 @@ public void SetFunctionInactive(FunctionDefinitionNode functionDef)
             // Update the functiond definition in the codeblocks
             int hash = CoreUtils.GetFunctionHash(functionDef);
 
-            ProcedureNode procNode = null;
+            List<ProcedureNode> procNodes = null;
 
             foreach (CodeBlock block in CodeBlockList)
             {
                 // Update the current function definition in the current block
-                procNode = block.procedureTable.Procedures.FirstOrDefault(p => p.HashID == hash);
-                int index = procNode == null ? Constants.kInvalidIndex: procNode.ID; 
-                if (Constants.kInvalidIndex == index)
-                    continue;
-
-                procNode.IsActive = false;
-
-                // Remove staled graph nodes
-                var graph = block.instrStream.dependencyGraph;
-                graph.GraphList.RemoveAll(g => g.classIndex == ClassIndex && 
-                                               g.procIndex == index);
-                graph.RemoveNodesFromScope(Constants.kGlobalScope, index);
-
-                // Make a copy of all symbols defined in this function
-                var localSymbols = block.symbolTable.symbolList.Values
-                                        .Where(n => 
-                                                n.classScope == Constants.kGlobalScope 
-                                             && n.functionIndex == index)
-                                        .ToList();
-
-                foreach (var symbol in localSymbols)
+                procNodes = block.procedureTable.Procedures.Where(p => p.HashID == hash).ToList();
```

Comment:
so what situation causes a procedure in the procedureTable to somehow have the same hashID, isn't it a hash table? How can their be more than one?

Reply:
`Procedures` is a list, not a hashtable.

Reply:
As long as the function name and the argument types and strings are the same, the same hash ID is generated. So if a function is deleted, then re-created, say if you delete a CBN and undo the deletion, a new entry for the same function with the same hash ID will be added, after the previous one has been marked inactive. The new one will be active.

Reply:
1. It's surprising that it's not a table at all... any idea why it's implemented with a list?
2. Does it make sense to use some query method on the procedureTable instead of accessing the list direclty?
3. Did you see this case come up or it's just defensive coding?

Reply:
oh interesting.. so the procedure table will fill up with inactive functions - should it be cleaned up at some point? If we don't already do that?

Reply:
@pinzart90 had the same question. I don't have an answer to why a list was chosen and not a dictionary/hash table. I just think it was a poor choice of data structure. We do this even for the instruction stream where that is another list that keeps growing. Deleting instead of maintaining an ever-expanding list (and marking stale items inactive) is definitely the way to go but it's a bigger change and probably outside the scope of this PR. There needs to be a separate spike/epic to optimize these data structures.

Reply:
The reason its name contains the word "table" is probably because it is aimed at representing that concept of a function table in traditional compiler design.